support/README.md

```diff
@@ -0,0 +1,36 @@
+# Support tools
+
+A collection of CLI tools meant to support and automate various aspects of
+developing Envoy, particularly those related to code review. For example,
+automatic DCO signoff and pre-commit format checking.
+
+## Usage
+
+To get started, you need only navigate to the Envoy project root and run:
+
+```bash
+./support/bootstrap
+```
+
+This will set up the development support toolchain automatically. The toolchain
+uses git hooks extensively, copying them from `support/hooks` to the `.git`
+folder.
+
+The commit hook checks can be skipped using the `-n` / `--no-verify` flags, as
+so:
+
+```bash
+git commit --no-verify
```

Comment:
Before getting into too much review depth, I'm wondering if we really want this to be opt-out on commit vs. push. My common workflow (and I'm sure is the case for many devs) is that we commit early-and-often locally, without caring about format or DCO (which happens via my git aliases anyway). I later squash down in a rebase and then push. I rather that any additional latency is added on the remote push rather than the local commit.

Of course, these are just my preferences and I don't need to actually use these hooks, but I think they may be shared by others as well.